src/gsNurbs/gsTensorNurbs.h

```diff
@@ -369,7 +369,38 @@ class gsTensorNurbs : public gsGeoTraits<d,T>::GeometryBase
         this->basis().swapDirections(i,j);
     }
 
-
+    /// Splits the geometry either two parts in direction \a dir, or if \a dir = -1
+    /// in 2^d parts, by calling splitAt() for each direction.
+    /// The function automatically searches for the midpoint the corresponding knot vector.
+    std::vector<gsGeometry<T>*> uniformSplit(index_t dir = -1) const
+    {
+        // We use the simple fact that a NURBS function in R^d is the central probjection
+        // of a spline function in R^{d+1} into R^d.
+        //
+        // Create a B-spline in R^{d+1} and split it
+        gsRationalBasis< gsTensorBSplineBasis<d,T> > * basis
```

Comment:
I think this can be replaced as
Basis & basis = this->basis(); //Basis is here gsTensorNurbsBasis

Reply:
The GISMO_ENSURE that follows can be omitted: in this context the basis is of the correct type..